### ISSUE TITLE
Flatpak build

### ISSUE THREAD

**Issue by derekstavis** [artemanufrij/playmymusic#55]
Heya!!

Looks like this music player rocks!! I have tried building myself with vanilla GNOME on Arch Linux, but despite from having all dependencies installed I couldn't achieve a build.

Thinking about this, it would be nice to have this music player distributed as a flatpak, so other distros could also enjoy it 👍

**Comment by bil-elmoussaoui** [comment]
Hey, 
You couldn't achieve a successfully build because the applications depend on cmake-elementary. The application should really add those directly to the source code instead :( or move to using meson. 

Anyway, there should be a Flatpak package pretty soon! stay tund :+1:

**Comment by bil-elmoussaoui** [comment]
@artemanufrij I was able to successfully build a flatpak package see 
https://github.com/flathub/flathub/pull/229
There are few small issues that should be fixed in the next release of your application so my PR can be merged upstream
1 - The appdata is not valid. You can run `appstream-util validate myappdatafile.appdata.xml` and you will get the list of the issues that should be fixed
2 - The headerbar looks a bit huge in other platforms (even if your application was meant to be used in elementary only..) I think it shouldn't be too complicated to fix it ;) just check Vocal app.
![screenshot from 2018-01-15 23-01-50](https://user-images.githubusercontent.com/7660997/34963311-19fa070e-fa48-11e7-8f94-507a2a5a0c04.png)
3 - The media keys don't work as expected, either I didn't use the right dbus name, or it's something wrong with the media keys (works with older gnome settings versions but not the recent ones). I will let you know about this one.

The rest is not yet defined :p but I will keep you up to date :)

Once this one is done, I will package the playmyvideos too :)

**Comment by artemanufrij** [comment]
@bil-elmoussaoui thank you for your feedback. I will check it tomorrow

**Comment by bil-elmoussaoui** [comment]
And yes, please include the cmake files that I have added in my PR https://github.com/flathub/flathub/pull/229/files to a cmake folder and apply the cmake.patch :)
This will avoid us both unneeded work! Or just try meson and never go back to the hell of cmake :p

**Comment by bilelmoussaoui** [comment]
Sorry for the long delay. I'm going to update my PR on flathub today.

**Comment by bilelmoussaoui** [comment]
Already on Flathub https://github.com/flathub/com.github.artemanufrij.playmymusic
Can be closed, please report any issues you have there

**Comment by bilelmoussaoui** [comment]
@artemanufrij Can you please try to run the application as a Flatapk and see if you have any issues or anything that can be fixed to improve the application

**Comment by bilelmoussaoui** [comment]
https://flathub.org/apps/details/com.github.artemanufrij.playmymusic

**Comment by artemanufrij** [comment]
@bilelmoussaoui thank you for your great support!
